docs/pages/blog.tsx

```diff
@@ -0,0 +1,441 @@
+                  <Box
+                    component="img"
+                    src={post.image}
+                    sx={{
+                      aspectRatio: '16 / 9',
+                      width: '100%',
+                      height: 'auto',
+                      objectFit: 'cover',
+                      borderRadius: '4px',
+                    }}
+                  />
+                )}
+                <PostPreview {...post} />
+              </Paper>
+            ))}
+          </Box>
+        </Section>
+        <Container
+          id="post-list"
+          sx={{
+            mt: { xs: -8, sm: -7 },
+            display: 'grid',
+            gridTemplateColumns: { md: '1fr 380px' },
+            columnGap: 8,
+          }}
+        >
+          <Typography
+            component="h2"
+            color="text.primary"
+            variant="h5"
+            fontWeight="700"
+            sx={{ mb: { xs: 1, sm: 2 }, mt: 8 }} // margin-top makes the title appear when scroll into view
+          >
+            Posts{' '}
+            {Object.keys(selectedTags).length ? (
+              <span>
+                about{' '}
+                <Typography component="span" variant="inherit" color="primary" noWrap>
+                  &quot;{Object.keys(selectedTags)[0]}&quot;
+                </Typography>
+              </span>
+            ) : (
+              ''
+            )}
+          </Typography>
+          <Box sx={{ gridRow: 'span 2' }}>
+            <Box
+              sx={{
+                position: 'sticky',
+                top: 100,
+                alignSelf: 'start',
+                mb: { xs: 2, sm: 8 },
+                mt: { xs: 2, sm: 8 }, // margin-top makes the title appear when scroll into view
+                p: 2,
+                borderRadius: 1,
+                border: '1px solid',
+                background: (theme) =>
+                  theme.palette.mode === 'dark'
+                    ? alpha(theme.palette.primaryDark[700], 0.2)
+                    : 'rgba(255, 255, 255, 0.2)',
+                borderColor: (theme) =>
+                  theme.palette.mode === 'dark'
+                    ? theme.palette.primaryDark[700]
+                    : theme.palette.grey[200],
+              }}
+            >
+              <Typography color="text.primary" fontWeight="500" sx={{ mb: 2 }}>
+                Filter by subject
+              </Typography>
+              <Box sx={{ display: 'flex', gap: 1, flexWrap: 'wrap' }}>
+                {allTags.map((tag) => {
+                  const selected = !!selectedTags[tag];
+                  return (
+                    <Chip
+                      key={tag}
+                      variant={selected ? 'filled' : 'outlined'}
+                      {...(selected
+                        ? {
+                            label: tag,
+                            onDelete: () => {
+                              const postList = document.getElementById('post-list');
```

Comment:
Could this be inside an effect? It could scroll every time that the `tags` change.

Using `document.getElementById` directly is not very React-ish. A ref should be used instead.

Reply:
> Could this be inside an effect

I don't think so. What I want is to scroll when `onDelete` is called

> Using document.getElementById directly is not very React-ish

I will change to ref instead.